play-json/js-native/src/main/scala/StaticBindingJsNative.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2009-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.api.libs.json
+
+import java.io.InputStreamReader
+
+private[json] object StaticBindingJsNative {
```

Comment:
The code in this file was move from `StaticBinding` in the js implementation.

Reply:
I'm just curious here - what's the point of this restructure?

Reply:
> Ah, I think I see - you're making the underlying functionality common, so that both `js` and `native` `StaticBinding` implementations can access this code?

Exactly, to avoid copy-pasting a lot of code, I put the common code here so it's shared and called back and forth from the Native and JS implementations.

Reply:
A small nit then - I found the `js-native` and `StaticBindingJsNative` names a bit confusing. The name doesn't really convey the intent, and isn't intuitive for further possible expansions.

Reply:
the `js-native` directory is automatically added by `sbt-crossproject`.
About `StaticBindingJsNative` I just added the platforms names to the original file name. If you have a better suggestion I change it. I don't have any better name in mind.. maybe `StaticBindingNonJvm` ?

Reply:
Ah, I had a feeling it was sbt plugin shenanigans causing the package name. Makes sense, given that the subprojects aren't explicitly depending on one another in `build.sbt`.

I think it's really a nit though, I hate to delay for this. Let's leave it as is, and leave better naming for later. This isn't a public API, so it shouldn't affect very many users.

I will say though, `StaticBindingNonJvm` or `StaticBindingNonJvmShared` is a bit more descriptive.